boot: espressif: integrate Espressif Port with Zephyr sysbuild system

almir-okato opened the pull request:
This PR enables building MCUboot Espressif Port through Zephyr's sysbuild for Espressif chips.
- Espressif's CMakeLists script content was reorganized in order to separate Zephyr's specifics ways to add and configure target libraries to its build system.
- Kconfig was added (still only used when building from Zephyr sysbuild).
- Extra .conf files for the SoCs were added in order to disable peripheral configurations used by Zephyr that may conflict on linkage as Espressif Port has its own way to deal with. Espressif Port standalone build is not affected by this addition.

Note that these changes affects only the Espressif Port build, nothing was changed on Zephyr Port of MCUboot.

LucasTambor committed:
boot: espressif: integrate Espressif Port with Zephyr sysbuild system

Enable building MCUboot Espressif Port through Zephyr's sysbuild for
Espressif chips.

This commit creates boot/espressif/zephyr directory and files
required by Zephyr's build system to build Espressif Port through
sysbuild or as "out-of-tree" project.
It also moves shared parts from the main espressif/CMakeLists.txt
script.

Signed-off-by: Lucas Tamborrino <lucas.tamborrino@espressif.com>
Signed-off-by: Almir Okato <almir.okato@espressif.com>

almir-okato committed:
boot: espressif: reorganize cmake build script file

Signed-off-by: Almir Okato <almir.okato@espressif.com>

nordicjm commented on boot/espressif/CMakeLists.txt:
```diff
@@ -7,48 +7,57 @@ cmake_policy(SET CMP0109 NEW)
+
+# Set MCUboot Espressif Port configuration file
+if(NOT DEFINED MCUBOOT_CONFIG_FILE)
+    set(MCUBOOT_CONFIG_FILE "${ESPRESSIF_PORT_DIR}/port/${MCUBOOT_TARGET}/bootloader.conf")
```
cmake indent is 2 spaces

almir-okato replied:
Fixed

nordicjm commented on boot/espressif/CMakeLists.txt:
```diff
@@ -7,66 +7,83 @@ cmake_policy(SET CMP0109 NEW)
-set(APP_NAME mcuboot_${MCUBOOT_TARGET})
-set(APP_EXECUTABLE ${APP_NAME}.elf)
-set(EXPECTED_IDF_HAL_VERSION "5.1.4")
+    find_package(Zephyr REQUIRED HINTS $ENV{ZEPHYR_BASE})
```
bit confused here, if sysbuild is set then this should already be from zephyr scope, why is this finding zephyr when it is running from zephyr?

almir-okato replied:
This is needed because the project is being included as an external Zephyr project from sysbuild (https://github.com/zephyrproject-rtos/zephyr/pull/87710).
MCUboot Zephyr port does the same to include the Zephyr environment.

nordicjm commented on boot/espressif/Kconfig:
```diff
@@ -0,0 +1,363 @@
+config ESP_USE_MBEDTLS
+	bool
+	# Hidden option
+	default n
```
```suggestion
```

almir-okato replied:
Fixed

nordicjm commented on boot/espressif/Kconfig:
```diff
@@ -0,0 +1,363 @@
+config ESP_USE_TINYCRYPT
+	bool
+	# Hidden option
+	default n
```
```suggestion
```

almir-okato replied:
Fixed

nordicjm commented on boot/espressif/Kconfig:
```diff
@@ -0,0 +1,363 @@
+
+	config ESP_SINGLE_APPLICATION_SLOT
+		bool "Single slot bootloader"
+		default n
```
```suggestion
```
remove all `default n`

almir-okato replied:
Fixed

nordicjm commented on boot/espressif/Kconfig:
```diff
@@ -0,0 +1,363 @@
+		uploading a new application overwrites the one that previously
+		occupied the area.
+
+	choice ESP_SIGN_TYPE
```
do not indent options in Kconfig

nordicjm commented on boot/espressif/Kconfig:
```diff
@@ -0,0 +1,363 @@
+		with the public key information will be written in a format expected by
+		MCUboot.
+
+	#config BOOT_VALIDATE_SLOT0
```
can be removed?

almir-okato replied:
Removed

nordicjm commented on boot/espressif/Kconfig:
```diff
@@ -0,0 +1,363 @@
+        help
+            SPI flash size, in megabytes
+
+        config ESP_FLASHSIZE_1MB
```
why isn't this coming from dts?

almir-okato replied:
Fixed

nordicjm commented on boot/espressif/Kconfig:
```diff
@@ -0,0 +1,363 @@
+
+    config ESP_FLASH_SIZE
+        string
+        default "1MB" if ESP_FLASHSIZE_1MB
```
as above

almir-okato replied:
Fixed

nordicjm commented on boot/espressif/Kconfig:
```diff
@@ -0,0 +1,363 @@
+
+    config ESP_BOOTLOADER_OFFSET
+        hex
+		default $(dt_node_reg_addr_hex,$(DT_BOOT_PARTITION_PATH)) if BUILD_FROM_ZEPHYR
```
indent here also goes weird

almir-okato replied:
Fixed

nordicjm commented on boot/espressif/Kconfig:
```diff
@@ -0,0 +1,363 @@
+        default 43
+
+    config ESP_CONSOLE_UART_RX_GPIO
+        int "UART RX GPIO pin"
```
should come from dts

almir-okato commented:
@nordicjm I've changed what you pointed, could you please take a look again?

marekmatej commented:
hi @almir-okato, as you are using MCUboot EP `bootloader.ld`, maybe it is a good idea to use `memory.h` bootloader base address calculations to avoid segment collisions in the future when loading SRAM extensive images.

marekmatej reviewed (changes requested):
not sure whether linker issue would be solved here on in Zephyr PR, so I'm putting -1 here

Commit:
boot: espressif: Fix build problem with mbedtls

Signed-off-by: Danill <daniil.suhih@midomis.ru>
Signed-off-by: Almir Okato <almir.okato@espressif.com>

almir-okato committed:
boot: espressif: fix total_len unused variable warning

Declare total_len variable only when DEBUG log level is enabled.

Signed-off-by: Almir Okato <almir.okato@espressif.com>

almir-okato committed:
boot: espressif: add .xt.prop to the compiled ELF file

Add sections used by the debugger and disassembler to
get more information about raw data present in the code.

Signed-off-by: Almir Okato <almir.okato@espressif.com>

almir-okato committed:
boot: espressif: split dram_seg into dram_loader_seg memory segment

Create another segment from dram_seg, dram_loader_seg, to place
objects accessed by code in iram_loader_seg as it must not be
overlapped while loading main application.
Also reorganize the linker scripts to improve these memory spaces.

Signed-off-by: Almir Okato <almir.okato@espressif.com>
